api/specs.proto

```diff
@@ -336,6 +334,15 @@ message NetworkSpec {
 	bool ingress = 7;
 }
 
+// NetworkSpec specifies user defined network parameters.
+message NetworkSpec {
+	Annotations annotations = 1 [(gogoproto.nullable) = false];
+
+	oneof spec {
```

Comment:
`spec` feels a little redundant with the top-level message name. We end up with expressions involving `network.Spec.Spec`. Is there a more descriptive name for the oneof that captures the differences between cnm and cni?